### PR TITLE
Confs

### PR DESCRIPTION
**Pull request by martindurant**
This may be a more robust way to find the namenode/port. We still need to consider other places that the HDFS configurations may sit. Of course, for clients external to the cluster, both will have to be provided every time. We could optionally look for some specific environment variable, but I'd rather not.

### CODE REVIEW

File: hdfs3/core.py [hunk 1]
```diff
@@ -8,21 +8,55 @@
...
+                try:
+                    val = float(val)
+                except ValueError:
+                    pass
+            if val == 'false':
+                val = False
+            if val == 'true':
+                val = True
+            conf[key] = val
+    return conf
```
Comment (mrocklin) [inline hdfs3/core.py]: Is there a way to test this logic to ensure that it remains accurate?  I'm afraid of people making breaking changes in the future.

**Comment by martindurant** [comment]
Attempt to solve #17

**Comment by mrocklin** [comment]
If you add the term Fixes #17  (specifically the word fixes) then when this gets merged the other issue with close.

File: hdfs3/core.py (hunk 1 above)
Comment (martindurant) [inline hdfs3/core.py]: We can set a specific conf file and anticipate the correct parsing, but total generality is hard, could compare against a true XLM parsing (e.g., if the start- and end-tag are are different lines, this code breaks, but can the java XLM parser cope with that?).

Comment (mrocklin) [inline hdfs3/core.py]: A simple test to exercise functionality in the common case should be a good start.  Very simple tests identify most usage breaks.  More rigorous tests are great too of course if they can be done in moderate time.  I'm only really asking for a simple test though in this case.

**Comment by martindurant** [comment]
fixes #17

**Pull request merged**